src/VscPreviewExtension/src/DfmCore.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { workspace, window, ExtensionContext, Uri }from "vscode";
+import * as child_process from "child_process";
+import { MDDocumentContentProvider } from "./MDDocumentContentProvider";
+
+// Create a child process(DfmRender) by "_spawn" to render a html
+export class PreviewCore {
+    public _isFirstTime: boolean;
+    public _provider: MDDocumentContentProvider;
+
+    private _spawn: child_process.ChildProcess;
+    private _waiting: boolean;
+    private _previewContent: string;
+    private _isMultipleRead = false;
+    private _documentUri: Uri;
+    private ENDCODE = 7; // '\a'
+
+    constructor(context: ExtensionContext) {
+        let extpath = context.asAbsolutePath("./DfmParse/PreviewCore.exe");
+        this._spawn = child_process.spawn(extpath);
+        if (!this._spawn.pid) {
+            window.showErrorMessage("Error:DfmProcess lost!");
+            return;
+        }
+        this._waiting = false;
+        this._provider = new MDDocumentContentProvider(context);
+        let that = this;
+
+        this._spawn.stdout.on("data", function (data) {
+            // The output of child process will be cut if it is too long
+            let dfmResult = data.toString();
+            if (dfmResult.length !== 0) {
+                let endCharCode = dfmResult.charCodeAt(dfmResult.length - 1);
+                if (!that._isMultipleRead) {
```

Comment:
what about:
if (that._isMultipleRead) {that._previewContent += dfmResult;} else {that._previewContent = dfmResult;}
that.isMultipleRead = !(endCharCode === that.ENDCODE);
if (!_isMultipleRead) {that._provider.update(...)}